server/src/main/java/io/crate/analyze/TableInfoToAST.java

```diff
@@ -299,31 +304,58 @@ private Optional<ClusteredBy<Expression>> createClusteredBy(DocTableInfo docTabl
     private GenericProperties<Expression> extractTableProperties() {
         // WITH ( key = value, ... )
         Map<String, Expression> properties = new HashMap<>();
+        String numberOfReplicasKey = TableParameters.stripIndexPrefix(NumberOfReplicas.SETTING.getKey());
         if (tableInfo instanceof DocTableInfo docTable) {
             Expression numReplicas = new StringLiteral(docTable.numberOfReplicas());
-            properties.put(TableParameters.stripIndexPrefix(NumberOfReplicas.SETTING.getKey()), numReplicas);
+            properties.put(numberOfReplicasKey, numReplicas);
             properties.put("column_policy", new StringLiteral(docTable.columnPolicy().lowerCaseName()));
         }
-        Map<String, Object> tableParameters = TableParameters.tableParametersFromIndexMetadata(tableInfo.parameters());
-        for (Map.Entry<String, Object> entry : tableParameters.entrySet()) {
-            properties.put(
-                TableParameters.stripIndexPrefix(entry.getKey()),
-                Literal.fromObject(entry.getValue())
-            );
+        Settings parameters = tableInfo.parameters();
+        Map<String, Setting<?>> supportedSettings = TableParameters.TABLE_CREATE_PARAMETER_INFO.supportedSettings();
+        for (var entry : supportedSettings.entrySet()) {
+            String settingName = entry.getKey();
+            if (settingName.equals(numberOfReplicasKey)) {
+                // already handled above
+                continue;
+            }
+            Setting<?> setting = entry.getValue();
+            if (setting instanceof Setting.AffixSetting<?> affixSetting) {
+                String prefix = affixSetting.getKey();
+                Set<String> namespaces = affixSetting.getNamespaces(parameters);
+                for (String namespace : namespaces) {
+                    String key = prefix + namespace;
+                    Object value = affixSetting.getConcreteSetting(key).get(parameters);
+                    properties.put(key, literalOfSettingValue(value));
+                }
+            } else if (parameters.hasValue(setting.getKey())) {
+                Object value = setting.get(parameters);
+                properties.put(settingName, literalOfSettingValue(value));
+            }
         }
         return new GenericProperties<>(properties);
     }
 
-    private List<Expression> expressionsFromReferences(List<Reference> columns) {
-        List<Expression> expressions = new ArrayList<>(columns.size());
+    private static Literal literalOfSettingValue(Object value) {
```

Comment:
I've on my mind to have some better solution for this eventually because we generally have a bit of a mismatch between the core setting value types and the sql data types - that discrepancy is also noticeable in `SET [GLOBAL / SESSION]` statements where the value is a expression with a `DataType<?>` value but we internally use `Setting<?>` instances to represent them.

But it's currently not my main focus, so this will have to wait for another time